### DISCUSSION TITLE
test_dnn failing

### DISCUSSION THREAD

**Discussion by QiQi-Guo** [davisking/dlib#2931] [Q&A]
After build dlib and tests, I'm getting this error when running ./dtest --runall

Running test_dnn /                                                             

!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
!!!!!!!!!!!!!!!!!!!!!!!!!!!!! TEST FAILED: test_dnn !!!!!!!!!!!!!!!!!!!!!!!!!!!!!
!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!

Failure message from test: 

Error occurred at line 2011.
Error occurred in file /home/gq/dlib/dlib/test/dnn.cpp.
Failing expression was res.
Gradient error in parameter #0.  Relative error: 0.848535
expected derivative: -3.19285
output derivative:   -0.483604
iteration:           0

                                                                               

Testing Finished
Total number of individual testing statements executed: 470123422
Number of failed tests: 1
Number of passed tests: 143

**Comment by arrufat** [comment]
It's a known issue in the `layer_norm` gradients: https://github.com/davisking/dlib/issues/2902
